clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -50,10 +50,9 @@
         int maxKey = -1;
         for (ApiKeys key : ApiKeys.values())
             maxKey = Math.max(maxKey, key.id);
-        ApiKeys[] idToType = new ApiKeys[maxKey + 1];
+        ID_TO_TYPE = new ApiKeys[maxKey + 1];
         for (ApiKeys key : ApiKeys.values())
-            idToType[key.id] = key;
-        ID_TO_TYPE = idToType;
+            ID_TO_TYPE[key.id] = key;
```

Comment:
The idea is to not mutate the map after assignment, so the previous code was fine.